frontend/src/components/object/ObjectTable.vue

```diff
@@ -173,34 +188,53 @@ const selectedFilters = (payload: any) => {
     }, {});
   lazyParams.value.filters = filters;
 };
+
+// Exposed to update datatable rows
+function updateDataTable() {
+  loadLazyData();
+}
+
+defineExpose({
+  updateDataTable
+});
 </script>
 
 <template>
   <div class="object-table">
     <DataTable
       ref="dt"
       v-model:value="tableData"
-      v-model:selection="selectedObjects"
+      v-model:selection="objectStore.selectedObjects"
       v-model:filters="filters"
+      :select-all="selectAll"
       lazy
       paginator
       :loading="loading"
       :total-records="totalRecords"
       data-key="id"
       class="p-datatable-sm"
       responsive-layout="scroll"
-      :rows="3"
+      :rows="10"
       :rows-per-page-options="[10, 20, 50]"
       sort-field="updatedAt"
       :sort-order="-1"
-      filter-display="row"
       :global-filter-fields="['name']"
       :first="first"
       update:filters
       update:page
       @page="onPage($event)"
       @sort="onSort($event)"
       @filter="onFilter($event)"
+      @select-all-change="
+        (e) => {
+          selectAll = e.checked;
+          if (e.checked) {
+            selectCurrentPage();
+          } else {
+            objectStore.setSelectedObjects([]);
+          }
+        }
+      "
```

Comment:
As Tim mentioned we may be able to remove this logic as we are now always selecting the entire payload.